aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -148,11 +152,24 @@ public void signUp(
             @NonNull final Consumer<AuthSignUpResult> onSuccess,
             @NonNull final Consumer<AuthException> onException
     ) {
+        Map<String, String> userAttributes = new HashMap<>();
+        Map<String, String> validationData = new HashMap<>();
+
+        if (options.getUserAttributes() != null) {
+            for (AuthUserAttribute attribute : options.getUserAttributes()) {
+                userAttributes.put(attribute.getKey().getKeyString(), attribute.getValue());
+            }
+        }
+
+        if (options instanceof AWSCognitoAuthSignUpOptions) {
+            validationData = ((AWSCognitoAuthSignUpOptions) options).getValidationData();
```

Comment:
`.putAll()`?

Reply:
validationData in SignUpOptions is immutable so I don't think this is necessary